crates/turborepo-lib/src/gitignore.rs

```diff
@@ -0,0 +1,141 @@
+#[cfg(unix)]
+use std::os::unix::fs::PermissionsExt;
+use std::{
+    fs::{File, OpenOptions},
+    io,
+    io::{BufRead, Write},
+};
+
+use turbopath::AbsoluteSystemPathBuf;
+
+const TURBO_GITIGNORE_ENTRY: &str = ".turbo";
+const GITIGNORE_FILE: &str = ".gitignore";
+
+pub fn ensure_turbo_is_gitignored(repo_root: &AbsoluteSystemPathBuf) -> Result<(), io::Error> {
+    let gitignore_path = repo_root.join_component(GITIGNORE_FILE);
+
+    if !gitignore_path.exists() {
+        let mut gitignore = File::create(gitignore_path)?;
+        #[cfg(unix)]
+        gitignore.metadata()?.permissions().set_mode(0o0644);
+        writeln!(gitignore, "{}", TURBO_GITIGNORE_ENTRY)?;
```

Comment:
We have [create_with_contents](https://github.com/vercel/turbo/blob/main/crates/turborepo-paths/src/absolute_system_path.rs#L177) that can combine the file creation and writing.